### PR TITLE
Add umask configuration for generated proxies

### PR DESCRIPTION
**Pull request by bcremer**
Depends on https://github.com/doctrine/common/pull/711.
Fixes https://github.com/doctrine/common/issues/710.

### CODE REVIEW

File: lib/Doctrine/ORM/Proxy/ProxyFactory.php [hunk 1]
```diff
@@ -71,10 +76,11 @@ class ProxyFactory extends AbstractProxyFactory
      * @param string                 $proxyNs      The namespace to use for the proxy classes.
      * @param boolean|int            $autoGenerate The strategy for automatically generating proxy classes. Possible
      *                                             values are constants of Doctrine\Common\Proxy\AbstractProxyFactory.
+     * @param int                    $umask        The umask that controls file permissions.
      */
-    public function __construct(EntityManagerInterface $em, $proxyDir, $proxyNs, $autoGenerate = AbstractProxyFactory::AUTOGENERATE_NEVER)
+    public function __construct(EntityManagerInterface $em, $proxyDir, $proxyNs, $autoGenerate = AbstractProxyFactory::AUTOGENERATE_NEVER, $umask = 0002)
     {
-        $proxyGenerator = new ProxyGenerator($proxyDir, $proxyNs);
+        $proxyGenerator = new ProxyGenerator($proxyDir, $proxyNs, $umask);
```
Comment (bcremer) [inline lib/Doctrine/ORM/Proxy/ProxyFactory.php]: This is a BC break and requires a version of doctrine/common with this patch included: https://github.com/doctrine/common/pull/711

File: lib/Doctrine/ORM/Configuration.php [hunk 2]
```diff
@@ -126,6 +126,33 @@ public function setProxyNamespace($ns)
...
 
     /**
+     * Returns the umask that controls file permissions for generated proxies.
+     *
+     * @return int
+     */
+    public function getProxyUmask()
+    {
+        return isset($this->_attributes['proxyUmask']) ?
+            $this->_attributes['proxyUmask'] : 0002;
```
Comment (bcremer) [inline lib/Doctrine/ORM/Configuration.php]: Do we want to maintain the default umask here? Or is it better to rely on the default value of the `$umask` parameter in the `ProxyGenerator`?

File: lib/Doctrine/ORM/Configuration.php [hunk 3]
```diff
@@ -126,6 +126,33 @@ public function setProxyNamespace($ns)
...
+    /**
+     * @since 2.5.6
+     *
+     * @param int $umask The umask that controls file permissions for generated proxies.
+     *
+     * @throws ORMException
+     */
+    public function setProxyUmask($umask)
+    {
+        if ( ! is_int($umask)) {
```
Comment (bcremer) [inline lib/Doctrine/ORM/Configuration.php]: Validation is Duplicated in `Doctrine\Common\Proxy\ProxyGenerator` - Is that a problem?

File: lib/Doctrine/ORM/Proxy/ProxyFactory.php (hunk 1 above)
Comment (bcremer) [inline lib/Doctrine/ORM/Proxy/ProxyFactory.php]: Any 	recommendations how to get the fix in v2.5.5 without breaking compatibility? I consider this more like a bug fix than a new feature given the nature of https://github.com/doctrine/common/issues/710.

Comment (Majkl578) [reply lib/Doctrine/ORM/Proxy/ProxyFactory.php]: Technically not on call-side, it's a constructor - parameter has default value and constructors don't follow LSP.

**Comment by Majkl578** [comment]
Hello, I'm going to close this PR as we will be removing Doctrine\Common\Proxy in favor of Ocramius\ProxyManager in Doctrine 3.0 and 2.x is closed feature-wise. Thank you for the PR.

**Comment by Majkl578** [comment]
Note: If you still need this to happen, please send PR/ticket to [ocramius/proxymanager](https://github.com/ocramius/proxymanager) and we can resurrect this PR afterwards.

**Pull request closed without merging**